src/ResultWriter/faultoutput.f90

```diff
@@ -630,30 +631,34 @@ SUBROUTINE calc_FaultOutput( DynRup_output, DISC, EQN, MESH, MaterialVal, BND, t
                 nDegFr2d = (DISC%Galerkin%nPoly + 1)*(DISC%Galerkin%nPoly + 2)/2
                 !projection of the rupture time on the basis functions
                 ALLOCATE(projected_RT(nDegFr2d))
+                ALLOCATE(phiAtPoint(2 * nDegFr2d))
                 projected_RT(:)=0.
 
                 DO jBndGP = 1,DISC%Galerkin%nBndGP
                   chi  = MESH%ELEM%BndGP_Tri(1,jBndGP)
                   tau  = MESH%ELEM%BndGP_Tri(2,jBndGP)
+                  call c_interoperability_TriDubinerP(phiAtPoint, chi, tau, DISC%Galerkin%nPoly)
                   DO iDegFr = 1, nDegFr2d
-                     call BaseFunc_Tri(phiT,iDegFr,chi,tau,DISC%Galerkin%nPoly,DISC)
                      projected_RT(iDegFr) = projected_RT(iDegFr) + &
-                        DISC%Galerkin%BndGaussW_Tet(jBndGP)*DISC%DynRup%output_rupture_time(jBndGP,iFace)*phiT
+                        DISC%Galerkin%BndGaussW_Tet(jBndGP) * &
+                        DISC%DynRup%output_rupture_time(jBndGP,iFace) * phiAtPoint(iDegFr)
                   ENDDO
                 ENDDO
                 DO iDegFr = 1,nDegFr2d
-                   projected_RT(iDegFr) =  projected_RT(iDegFr)/DISC%Galerkin%MassMatrix_Tri(iDegFr,iDegFr,DISC%Galerkin%nPoly)
+                   projected_RT(iDegFr) = projected_RT(iDegFr) * &
+                       c_interoperability_M2invDiagonal(iDegFr-1)
```

Comment:
are you sure about the -1?

Reply:
Yes, in Fortran you generally start counting from 1, in C/C++ generally from 0.